### DISCUSSION TITLE
Customize the nginx configuration for a Docker image

### DISCUSSION THREAD

**Discussion by robingenz** [dokku/dokku#6688] [Q&A] [answered]
First of all, thank you for this great project!

I am currently deploying a Docker image using the [`git:from-image`](https://dokku.com/docs/deployment/methods/image/) command. Now I need to customize the default nginx configuration to make the app work correctly. According to the [documentation](https://dokku.com/docs/networking/proxies/nginx/#customizing-the-nginx-configuration), I can use an `nginx.conf.sigil` file for this. However, this must be in the Docker image when deploying via `git:from-image`, see:

> For deploys via the `git:from-image` and `git:load-image` commands, the `nginx.conf.sigil` is extracted from the configured `WORKDIR` property of the image.

So my question is: Is there any option to define a custom nginx configuration without having to customize the Docker image?

**Comment by josegonzalez** [comment, marked as answer]
You could create a plugin that implements the `nginx-app-template-source` trigger and responds with a custom nginx.conf.sigil path for your app when it gets the `app-config` as the second argument. That will be respected [here](https://github.com/dokku/dokku/blob/bfe552ee43c9d2e4cd902545f6e09ca23026bfe4/plugins/nginx-vhosts/functions#L227) over the built-in one.

Outside of building a custom plugin, no, Dokku has no way to do what you want and it isn't on the roadmap to build such functionality.

**Reply by robingenz** [reply]
Thank you for your response!

Can the path that the plugin returns be a local path on the server or does it also have to be a path in the Docker image?

**Reply by josegonzalez** [reply]
Yep it would be a path to a file on disk readable by the dokku user.

**Reply by dewey** [reply]
@robingenz I had the same question and implemented a simple plugin to do just that. In case you still have use for that: https://github.com/dewey/dokku-nginx-override-by-app

**Reply by robingenz** [reply]
@dewey Thank you! I will take a look at it.

**Reply by jonathanmorris180** [reply]
@dewey Just wanted to chime in here and say thanks for building that! I was in the same boat, used your plugin, and it worked like a charm. 🙌
